python/ray/_private/runtime_env/conda.py

```diff
@@ -216,27 +291,31 @@ def setup(self,
                 extra_pip_dependencies = []
             conda_dict = inject_dependencies(conda_dict, _current_py_version(),
                                              extra_pip_dependencies)
+
             logger.info(f"Setting up conda environment with {runtime_env}")
             # It is not safe for multiple processes to install conda envs
             # concurrently, even if the envs are different, so use a global
             # lock for all conda installs.
             # See https://github.com/ray-project/ray/issues/17086
             file_lock_name = "ray-conda-install.lock"
             with FileLock(os.path.join(self._resources_dir, file_lock_name)):
-                conda_dir = os.path.join(self._resources_dir, "conda")
-                try_to_create_directory(conda_dir)
-                conda_yaml_path = os.path.join(conda_dir, "environment.yml")
-                with open(conda_yaml_path, "w") as file:
-                    # Sort keys because we hash based on the file contents,
-                    # and we don't want the hash to depend on the order
-                    # of the dependencies.
-                    yaml.dump(conda_dict, file, sort_keys=True)
-                conda_env_name = get_or_create_conda_env(
-                    conda_yaml_path, conda_dir, logger=logger)
-
-            if runtime_env.get("_inject_current_ray"):
-                conda_path = os.path.join(conda_dir, conda_env_name)
-                _inject_ray_to_conda_site(conda_path, logger=logger)
+                conda_yaml_file = os.path.join(self._resources_dir,
+                                               "environment.yml")
+                with open(conda_yaml_file, "w") as file:
+                    yaml.dump(conda_dict, file)
+
+                if conda_env_name in self._created_envs:
+                    logger.debug(f"Conda env {conda_env_name} already "
+                                 "created, skipping creation.")
+                else:
+                    create_conda_env(
+                        conda_yaml_file, prefix=conda_env_name, logger=logger)
+                    self._created_envs.add(conda_env_name)
+                os.remove(conda_yaml_file)
```

Comment:
in general we should have these cleanups in a `finally` block so they run even if the creation fails

Reply:
I see, makes sense